File: 3886-count-number-of-trapezoids-i/count-number-of-trapezoids-i.py
class Solution:
    
    def countTrapezoids(self, points: List[List[int]]) -> int:
        MOD = 10**9+7
        mp = {}
        
        for x,y in points:
            if y not in mp:
                mp[y]=1
            else:
                mp[y]+=1
        
        pairs=0
        res = 0
        for y in mp.keys():
            count = mp[y]
            pairs+=comb(count,2)

        for y in mp.keys():
            count = mp[y]
            curr = comb(count,2)
            res+= (curr * (pairs-curr))%MOD
        
        return (res//2)%MOD

File: 3886-count-number-of-trapezoids-i/count-number-of-trapezoids-i.cpp
class Solution {
public:
    const int MOD = 1e9+7;
    const long long INV2 = 500000004;
    int countTrapezoids(vector<vector<int>>& points) {
        unordered_map<int,int>mpp;
        for(auto it : points){
            mpp[it[1]]++;
        }
        long long sum=0;
        long long sq=0;
        for(auto [k,v]:mpp)
        {   
            long long comb = ((long long)v * (v - 1) / 2) % MOD;
            sum = (sum + comb)%MOD;
            sq = (sq + comb*comb) %MOD;
        }
        int res = (((sum*sum) %MOD - sq + MOD))%MOD;
        res = (res * INV2) % MOD;
        return (int)res;
    }
};